### ISSUE TITLE
p3 type instance not supported 

### ISSUE THREAD

**Issue by schemmy** [awslabs/cfncluster#262]
This type of instance is not included in current defaulted template_url, thus cannot be created or updated. 

The error is:
Parameter ComputeInstanceType failed to satisfy constraint: Must be a valid EC2 instance type, with support for HVM.

Can someone fix this, or let me know how to use my own template? Thanks.

**Comment by mohanasudhan** [comment]
We are working to release cfncluster 1.4.0 which will support all new instance type by default. We don't have exact timeline for release but I am hopeful we will have it in coming weeks. 

PR: https://github.com/awslabs/cfncluster/pull/240

In case if you want to use P3 asap via cfncluster, refer PR https://github.com/awslabs/cfncluster/pull/236/files for adding instance type in cloudformation template and can refer new cloudformation template using template_url flag http://cfncluster.readthedocs.io/en/latest/configuration.html#template-url

**Comment by barrettabel** [comment]
Hi @schemmy,


tl;dr:  You can do it, but it might be easier to wait a little.

I had to make this work for Amber and some MD packages now... The good thing is cfncluster has a good framework to customize!

You will need to add a custom template_url in your cluster config, and upload it to a bucket adding p3 references.. I 

i.e.

```template_url = https://s3-us-west-2.amazonaws.com/bla_bla/cfncluster.cfn.json```


Some of the patch so you get the idea:
```
755a756
>         "p3.2xlarge",
827a829
>         "p3.2xlarge",
1615a1618,1621
>         "Arch" : "64HVM",
>         "EBSOpt" : "True"
>       },
>       "p3.2xlarge" : {
```


Keep in mind the votla gpus only work with cuda 9 and fairly new drivers so you will need to update the AMIs.. I have some packer and ansible plays to help if you need to update the stock cfncluster AMIs.


There is a section in the documentation about using custom AMIs.. etc. take a look at that if you have not already..

Happy hunting!
